docs/architecture/adr-015-namespace-id-size.md

```diff
@@ -0,0 +1,212 @@
+# ADR 15: Namespace ID Size
+
+## Status
+
+Accepted
+
+## Changelog
+
+- 2023/2/17: initial draft
+- 2023/2/22: discussion notes
+- 2023/2/23: reorder content
+- 2023/2/28: NMT proof size
+- 2023/3/1: blob inclusion proof size
+- 2023/3/2: accepted
+
+## Context
+
+Namespace ID is currently an 8 byte slice. 8 bytes provides a maximum of 2^64 possible namespace IDs. In practice some namespace IDs are reserved for protocol use so the number of namespace IDs available to users is 2^56 - 2. Modifying the size of a namespace ID post-launch is a breaking change and has implications for the NMT, share encoding, etc. so we'd like to carefully consider the size of the namespace ID pre-launch.
+
+## Decision
+
+Increase the namespace ID size to 32 bytes.
```

Comment:
[Question] Can we have shares with different namespace versions in the same NMT?

Reply:
Depends on what change the new namespace version is introduced for but if a new namespace version is introduced to change something like non-interactive default rules then yes b/c both namespace version `0` and `1` will be of size 32 bytes (ID) + 1 byte (version) = 33 bytes.